### PR TITLE
DVCI-135 Handle Different V2 Codes

### PR DESCRIPTION
**Pull request by ssayer**
The API will catch V2 messages with zero or multiple PIDs and throw the appropriate error. This is then rendered by the controller as either a 404 (not found) or 422 (unprocessable entity).

Also addresses DVCI-160, so issuer should be populated now.

### CODE REVIEW

File: app/models/health_card.rb [hunk 1]
```diff
@@ -1,16 +1,28 @@
 class HealthCard
```
Comment (mickohanlon23) [inline app/models/health_card.rb:1]: Reece had mentioned not creating a new HealthCard class to avoid confusion with the HealthCard class from our library. Would this be a part of a separate PR?

**Review by mickohanlon23 — APPROVED** [review]
Patient found still works correctly, and 404 and 422 errors both work. V2 handling makes sense and throws the proper errors for no patient found and for multi-match. Next step would be getting the front-end to respond to both error cases. Looks good!

File: app/models/health_card.rb (hunk 1 above)
Comment (ssayer) [reply app/models/health_card.rb:1]: Yeah, I originally intended to do that, but after looking at it for a while I think we need to have discussion about it.
